test/oqs_test_libctx.c

```diff
@@ -0,0 +1,597 @@
+// SPDX-License-Identifier: Apache-2.0 AND MIT
+
+#include <openssl/core_names.h>
+#include <openssl/evp.h>
+#include <openssl/provider.h>
+#include <openssl/rand.h>
+#include <openssl/types.h>
+#include <string.h>
+
+#include "oqs_prov.h"
+#include "test_common.h"
+
+#define MAX_DUMMY_ENTROPY_BUFFERLEN 0x100000
+
+/** \brief The info about classical elements needed. */
+struct ClassicalInfo {
+    /** \brief Name label. */
+    const char *name;
+
+    /** \brief The public key length, in bytes. */
+    size_t pubkey_len;
+
+    /** \brief The private key length, in bytes. */
+    size_t privkey_len;
+
+    /** \brief The shared secret length, in bytes. */
+    size_t sec_len;
+
+    /** \brief The signature length, in bytes. */
+    size_t sig_len;
+};
+
+static const struct ClassicalInfo info_classical[] = {
+    {"p256", 65, 121, 32, 72},      {"SecP256r1", 65, 121, 32, 72},
+    {"p384", 97, 167, 48, 104},     {"p521", 133, 223, 66, 141},
+    {"bp256", 65, 122, 32, 72},     {"bp384", 97, 171, 48, 104},
+    {"rsa3072", 398, 1770, 0, 384}, {"pss3072", 398, 1770, 0, 384},
+    {"rsa2048", 270, 1193, 0, 256}, {"pss2048", 270, 1193, 0, 256},
+    {"ed25519", 32, 32, 0, 64},     {"ed448", 57, 57, 0, 114},
+    {"x25519", 32, 32, 32, 0},      {"X25519", 32, 32, 32, 0},
+    {"x448", 56, 56, 56, 0},
+};
+
+static OSSL_LIB_CTX *libctx = NULL;
```

Comment:
My question pertains to why this is required to be statically allocated and not passed as parameter?